modules/platform/nuxeo-elasticsearch/nuxeo-elasticsearch-core/src/main/java/org/nuxeo/elasticsearch/io/JsonESDocumentWriter.java

```diff
@@ -212,7 +217,12 @@ protected static void writeProperties(JsonGenerator jg, DocumentModel doc, Strin
         }
 
         for (Property p : properties) {
-            writer.writeProperty(jg, p);
+            try {
+                writer.writeProperty(jg, p);
+            } catch (ClassCastException e) {
+                throw new NuxeoException(
+                        String.format("writing JSON property failed on document: %s for property: %s", doc, p));
```

Comment:
Please keep the original exception as last argument when throwing a new one.

Reply:
Yes sorry about that. Fixed